wordpress/wp-content/plugins/sil-dictionary-webonary/webonary/Webonary_Cloud.php

```diff
@@ -72,48 +72,99 @@ private static function remoteFileUrl($path) {
+	}
+
+	private static function entryToDisplayXhtml($id, $entry) {	
 		//<div class="entry" id="ge5175994-067d-44c4-addc-ca183ce782a6"><span class="mainheadword"><span lang="es"><a href="http://localhost:8000/test/ge5175994-067d-44c4-addc-ca183ce782a6">bacalaitos</a></span></span><span class="senses"><span class="sensecontent"><span class="sense" entryguid="ge5175994-067d-44c4-addc-ca183ce782a6"><span class="definitionorgloss"><span lang="en">cod fish fritters/cod croquettes</span></span><span class="semanticdomains"><span class="semanticdomain"><span class="abbreviation"><span class=""><a href="http://localhost:8000/test/?s=&amp;partialsearch=1&amp;tax=9909">1.7</a></span></span><span class="name"><span class=""><a href="http://localhost:8000/test/?s=&amp;partialsearch=1&amp;tax=9909">Puerto Rican Fritters</a></span></span></span></span></span></span></span></div></div>
-		$id = self::convertGuidToId($entry->_id);
-		$mainHeadWord = '<span class="mainheadword"><span lang="' . $entry->mainHeadWord[0]->lang . '">'
-			. '<a href="' . get_site_url() . '/' . $id . '">' . $entry->mainHeadWord[0]->value . '</a></span></span>';
-				
-		$lexemeform = '';
-		if ($entry->audio->src != '') {
-			$lexemeform .= '<span class="lexemeform"><span><audio id="' . $entry->audio->id . '">';
-			$lexemeform .= '<source src="' . self::remoteFileUrl($dictionaryId . '/' . $entry->audio->src) . '"></audio>';
-			$lexemeform .= '<a class="' . $entry->audio->fileClass . '" href="#' . $entry->audio->id . '" onClick="document.getElementById(\'' . $entry->audio->id .   '\').play()"> </a></span></span>';
-		}
-	
-		// TODO: There can be multiple media files, e.g. Hayashi, one for lexemeform and another in pronunciations
-		$sharedgrammaticalinfo = '<span class="sharedgrammaticalinfo"><span class="morphosyntaxanalysis"><span class="partofspeech"><span lang="' . $entry->morphoSyntaxAnalysis->partOfSpeech[0]->lang . '">' . $entry->morphoSyntaxAnalysis->partOfSpeech[0]->value . '</span></span></span></span>';
-	
-		$sensecontent = '<span class="sensecontent"><span class="sense" entryguid="' . $id . '">'
-			. '<span class="definitionorgloss">';
-		foreach ($entry->senses[0]->definitionOrGloss as $definition)	{
-			$sensecontent .= '<span lang="' . $definition->lang . '">' . $definition->value . '</span>';
+		if (isset($entry->displayXhtml) && $entry->displayXhtml !== '') {
+			$displayXhtml = Webonary_Pathway_Xhtml_Import::fix_entry_xml_links($entry->displayXhtml);
+
+			// set image and audio src path to the cloud
+			if (preg_match_all('/src=\"(.*(?:\.jpg|.mp3))\"/iU', $entry->displayXhtml, $matches)) {
+				$baseUrl = self::remoteFileUrl($entry->dictionaryId) . '/';
+				foreach($matches[0] as $index => $src) {
+					$file = str_replace("\\", "/", $matches[1][$index]);
+					$displayXhtml = str_replace($src, 'src="' . $baseUrl . $file . '"', $displayXhtml);
+				}
+			}
+
+			// set semantic domains as links
+			if (preg_match_all(
```

Comment:
Same comment as above for all uses of `preg_match_all`.

Reply:
changed to explicitly check for === 1, since we need to act only if valid pattern is found. Added comment as well.